pkg/controller/daemon/daemoncontroller.go

```diff
@@ -429,10 +429,17 @@ func (dsc *DaemonSetsController) addNode(obj interface{}) {
 func (dsc *DaemonSetsController) updateNode(old, cur interface{}) {
 	oldNode := old.(*v1.Node)
 	curNode := cur.(*v1.Node)
-	if reflect.DeepEqual(oldNode.Labels, curNode.Labels) && reflect.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints) {
-		// If node labels and taints didn't change, we can ignore this update.
+
+	oldCond := v1helper.GetAvailableNodeCondition(oldNode.Status.Conditions)
+	curCond := v1helper.GetAvailableNodeCondition(curNode.Status.Conditions)
+
+	if reflect.DeepEqual(oldNode.Labels, curNode.Labels) &&
+		reflect.DeepEqual(oldNode.Spec.Taints, curNode.Spec.Taints) &&
+		v1helper.NodeConditionTypeEqual(oldCond, curCond) {
```

Comment:
Done; I filter by True with `NodeConditionEqual`, if `Status` is `False` or empty, ignore it.